lib/daru/io/importers/shared/request_log_analyzer_patch.rb

```diff
@@ -0,0 +1,48 @@
+module RequestLogAnalyzerPatch
+  module RequestLogAnalyzer::Source # rubocop:disable Style/ClassAndModuleChildren
+    # LogParser class, that reads log data from a given source and uses a file format
+    # definition to parse all relevent information about requests from the file
+    class LogParser
+      # Patch for the gem request-log-analyzer version 1.13.4 to combine the methods parse_file,
+      # parse_io and parse_line of the LogParser class. Assigns all the necessary instance
+      # variables defined in the above specified methods. Creates a request for each line of
+      # the file stream and stores the hash of parsed information in raw_list. Each element of
+      # parsed_list is an array of one parsed entry in log file.
```

Comment:
Ugh, now, I see a problem here (I thought that `LogParser` already _has_ method `parse_hash`, you just fixing something in it):

> combine the methods parse_file, parse_io and parse_line of the LogParser class

Why???? What's the point of combining three methods in one 40-lines mess?.. Why can't you use original methods as they were, just calling them?

Reply:
`parse_file` returns `nil` and so does `parse_io`. That's why I have tried fixing these methods in the earlier commits instead of making a new single method to avoid the rewriting as per your suggestion.

Reply:
Ugh. You should have spend a bit more time on understanding `LogParser` logic. From reading the code, I believe what you need is (without ANY patching) just call `LogParser#each_request`:

>  Reads the input, which can either be a file, sequence of files or STDIN to parse lines specified in the FileFormat. This lines will be combined into Request instances, that will be yielded.

...and, as it is aliased as `each`, and `Enumerable` is included, ALL you really need is:
```ruby
RequestLogAnalyzer::Source::LogParser
  .new(RequestLogAnalyzer::FileFormat.load(format))
  .map { converting Request into line for dataframe }
```
...and that would be completely it.